cmd/web/helpers.go

```diff
@@ -30,17 +31,23 @@ func (app *application) notFound(w http.ResponseWriter) {
 }
 
 func (app *application) render(w http.ResponseWriter, status int, page string, data *templateData) {
+	// Grab template from cache
 	ts, ok := app.templateCache[page]
 	if !ok {
 		err := fmt.Errorf("the template %s does not exist", page)
 		app.serverError(w, err)
 		return
 	}
 
-	w.WriteHeader(status)
-
-	err := ts.ExecuteTemplate(w, "base", data)
+	// Write to Buffer first to handle runtime errors
+	var buf bytes.Buffer
```

Comment:
Another way to do it is by: `buf := new(bytes.Buffer)` that would initialize buf on heap instead of stack like here. And you get pointer back so you don't need to use `&buf`.

Reply:
Either way works but this is preferred since `buf` is a short lived variable so living on a stack is perfectly okay.